packages/compass-saved-aggregations-queries/src/stores/aggregations-queries-items.ts

```diff
@@ -1,5 +1,102 @@
-function reducer(state = {}): typeof state {
+import { remote } from 'electron';
+import { join } from 'path';
+import fs from 'fs';
+import { AnyAction } from 'redux';
+import { FavoriteQueryCollection } from '@mongodb-js/compass-query-history';
+
+import { actions } from '../actions/aggregations-queries-actions';
+interface Item {
+  id: string;
+  name: string;
+  namespace: string;
+  lastModified: number;
+}
```

Comment:
I can imagine that we would need some kind of `type: 'aggregation' | 'query'` field here both to render those differently and to open the right tab when a user selects one.

Having a common "Item" type is very nice. Probably we don't want Query and Aggregation objects to share the same "space" for keys, they are unrelated objects, and technically, even if is not happening now, keys with the same name but different meanings may clash.

I'd suggest something like this:

```
interface AggregationItem extend Item {
   type: 'aggregation',  value: Aggregation
}
```

```
interface FindQueryItem extend Item {
   type: 'findQuery',  value: FindQuery
}
```

Reply:
Right now ended up having just [`Item` with `type`](https://github.com/mongodb-js/compass/commit/2448e4e2ea24612012807f581c21cbe5a1a31d5d) prop to differentiate between the two. Once we start implementing sort/filters, we will add data accordingly. CC @gribnoysup

Reply:
I'm not sure if the comment here points to a non-existent lines of code or I'm missing something, but the way `Item` type is defined right now in this PR makes sense to me